examples/slack-kit-example/README.md

```diff
@@ -1,38 +1,314 @@
+
+```
+"use client";
+
+import { KnockProvider, KnockSlackProvider } from "@knocklabs/react";
+
+export default function Providers({
+  userToken,
+  knockUserId,
+  tenant,
+  children,
+}: {
+  userToken: string;
+  knockUserId: string;
+  tenant: string;
+  children: JSX.Element;
+}) {
+  return (
+    <>
+      <KnockProvider
+        apiKey={process.env.NEXT_PUBLIC_KNOCK_CLIENT_ID!}
+        userId={knockUserId}
+        host={process.env.NEXT_PUBLIC_KNOCK_API_URL}
+        userToken={userToken}
+      >
+        {/*
+        The KnockProvider handles authentication with Knock, while the KnockSlackProvider
+        provides shared context to all Slack-related components.
+        Both are required for Slack-related apps.
+        */}
+        <KnockSlackProvider
+          knockSlackChannelId={process.env.NEXT_PUBLIC_KNOCK_SLACK_CHANNEL_ID!}
+          tenant={tenant}
+        >
+          {children}
+        </KnockSlackProvider>
+      </KnockProvider>
+    </>
+  );
+}
+```
+
+To create some of the necessary authentication data, like `userToken` and `userId`, we implement the `Providers` component inside of the root `layout.tsx` file. In this code, which runs on the server, we also add some additional grants to the `userToken` JWT to allow the user to interact with the Slack resources stored in Knock. You can read more about these [security and authentication steps in the docs](https://docs.knock.app/in-app-ui/security-and-authentication).
+
+**Note: this is necessary because the user in this context is an end user in your application who does not have access to Knock as a [member of the account](https://docs.knock.app/manage-your-account/managing-members). Therefore, these grants provide them elevated privileges to operate on specific resources using the API.**
+
+```
+import "@knocklabs/react/dist/index.css";
+import jwt from "jsonwebtoken";
+
+import Providers from "./components/providers";
+import "./global.css";
+import { getAppDetails } from "./lib/app-details";
+
+const { userId, tenant, collection, objectId } = getAppDetails();
+
+const currentTime = Math.floor(Date.now() / 1000);
+const expireInSeconds = 60 * 60;
+const signingKey = process.env.KNOCK_SIGNING_KEY!;
+
+const userToken = signingKey
+  ? jwt.sign(
+      {
+        sub: userId,
+        iat: currentTime,
+        exp: currentTime + expireInSeconds,
+        grants: {
+          [`https://api.knock.app/v1/objects/$tenants/${tenant}`]: {
+            "slack/channels_read": [{}],
+          },
+          [`https://api.knock.app/v1/objects/${collection}/${objectId}`]: {
+            "channel_data/read": [{}],
+            "channel_data/write": [{}],
+          },
+        },
+      },
+      signingKey,
+      {
+        algorithm: "RS256",
+      },
+    )
+  : "secretOrPrivateKey";
+
```

Comment:
we probably want to replace this with chris' new SDK right? https://github.com/knocklabs/knock-node/pull/56

Reply:
Yeah, looks like that's still open, so I'll loop back pre-release and update this